java/src/org/openqa/selenium/grid/data/DefaultSlotMatcher.java

```diff
@@ -93,25 +84,21 @@ public boolean matches(Capabilities stereotype, Capabilities capabilities) {
 
   private Boolean initialMatch(Capabilities stereotype, Capabilities capabilities) {
     return stereotype.getCapabilityNames().stream()
-        // Matching of extension capabilities is implementation independent. Skip them
-        .filter(name -> !name.contains(":"))
-        // Platform matching is special, we do it later
-        .filter(name -> !"platformName".equalsIgnoreCase(name))
-        .map(
-            name -> {
-              if (capabilities.getCapability(name) instanceof String) {
```

Comment:
The implementation of this conditional fails to consider that the corresponding stereotype capability could have a non-String value (including `null`), which would produce unexpected behavior. The revised implementation will only perform the case-insensitive comparison if the capability in the session request and the node stereotype both have String values.